src/disco/containers/DiscoPane.js

```diff
@@ -109,7 +135,42 @@ export function mapStateToProps(state) {
   };
 }
 
+export function mapDispatchToProps(dispatch, { _config = config } = {}) {
+  if (_config.get('server')) {
+    return {};
+  }
+  return {
+    handleGlobalEvent(e) {
+      const { id, type, needsRestart } = e;
+      const payload = { guid: id, needsRestart };
```

Comment:
I think it would be nice if this mostly moved into `addonManager`. So this could be:

```js
componentDidMount() {
  addonManager.addChangeListener(handleGlobalEvent);
}

return {
  handleGlobalEvent(type, payload) {
    dispatch({type, payload});
  }
};